### ISSUE TITLE
10$: Improve warning for Unicode control codes

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#892]
The following code contains U+202E RIGHT-TO-LEFT OVERRIDE before "6776". This is interpreted as the start of an identifier, and quick-lint-js confusingly reports E0057:
```js
transferBalance(5678,‮6776,4321‬,"USD");
```
```
hello.js:1:22: error: character is not allowed in identifiers [E0011]
hello.js:1:34: error: character is not allowed in identifiers [E0011]
hello.js:1:34: error: unexpected identifier in expression; missing operator before [E0147]
hello.js:1:1: warning: use of undeclared variable: transferBalance [E0057]
hello.js:1:22: warning: use of undeclared variable: 6776 [E0057]
hello.js:1:34: warning: use of undeclared variable:  [E0057]
```

There's also U+202C POP DIRECTIONAL FORMATTING after "4321", which also leads to E0057 (but it's harder to see in editors).

We should special case these invisible code points and report a different, more helpful diagnostic instead of just E0057. Perhaps we should skip the variables entirely in the variable-analyzer because we already reported E0011.
